File: c/basic/divisible_by_5.c
/*
 * Compile with: 
 *   gcc -Wall -o divisible_by_5 divisible_by_5.c
 * Run ./divisible_by_5
 */
#include <stdio.h>

int num1;
int num2;
const char QUIT = 'q';
char input[50];

int main(){
    
  while(1) {
    printf("Enter two numbers (first less than second):");
    
    fgets(input, 50, stdin);
    char quit;
    sscanf(input, "%c", &quit);
    if (quit == QUIT) {
      printf("Goodbye");
      return 0;
    } else {
      sscanf(input, "%d %d", &num1, &num2);
      int result = ((num2 - num1)/5) + 1;
      printf("There are %d multiples of 5 in this range\n", result);
    }
  }
  return 0;
}

File: ruby/basic/divisible_by_5.rb
loop do
  puts "Enter two numbers, first larger than second"
  input = gets.chomp
  if input == 'q'
    puts "Goodbye"
    break
  else
    num1, num2 = input.split(" ").map { |char| char.to_i }
    result = ((num2 - num1)/5) + 1

    puts "There are #{result} multiples of 5 in this range"
  end
end
